### ISSUE TITLE
Feature request

### ISSUE THREAD

**Issue by tigrazone123** [AdamYuan/VkAdypt#1]
Hello!
Did you plan to implement:
* HDRI light
* area lights
* full path tracing
?
It can be useful

**Comment by AdamYuan** [comment]
Yes, these features are in my TODO list, but may not be implemented in the near future.
Currently I'm working on parallel SBVH construction.

**Comment by tigrazone123** [comment]
Did you seen https://github.com/jbikker/bvh_article/? 
This article series have fast bvh methods

**Comment by AdamYuan** [comment]
It seems the article only described a naive single-threaded recursive binned SAH BVH constructor. The method didn't make use of multi-core CPU power, and the naive SAH BVH is inferior to SBVH in quality (ray tracing performance).
My parallel SBVH construction system aims to build high-quality SBVH as well as fully utilize CPU power, which is far more complex.

These are some of the references:
https://www.nvidia.in/docs/IO/77714/sbvh.pdf
https://diglib.eg.org/bitstream/handle/10.2312/pgv20161179/021-030.pdf 
http://www.sci.utah.edu/~wald/Publications/2010///LRB_BVH/download//LRB-BVH.pdf

**Comment by tigrazone** [comment]
Good thing is Blas, Tlas for instancing
